### DISCUSSION TITLE
Self hosting supabase with all the features.

### DISCUSSION THREAD

**Discussion by janaSunrise** [supabase/supabase#2218] [Questions] [answered]
Hello there 👋🏼 I'm a great fan of supabase, and I really enjoy it. 

I am interested in self hosting a node too for myself, and Personal projects locally, so I can have as much storage I want. But I noticed that It doesn't have the Frontend dashboard UI, Oauth features or such.

I would like to know, If it's Open Source, When will it be added, and Will it be even added?

The features I wish to see in self hosting are
- Dashboard UI
- OAuth features
- Functions
- And others, As included in the production version

Hoping to see these soon, Thank you! 😁 

**EDIT**: I found a related discussion, which had the same exact question for Frontend UI. Sorry for the disturbance I caused, and I would appreciate if this discussion can be closed / deleted. Sorry again! 😅

**Comment by btelles** [comment]
Gentle ping? :-)

**Comment by inian** [comment, marked as answer]
Hi @janaSunrise, all our backend services that we use in prod is in our self hosting setup [here](https://supabase.io/docs/guides/self-hosting). 

We haven't built functions out yet, but when we do, we will add it to our self hosting setup too. OAuth is part of our setup already. 

Dashboard UI is a work in progress and you can follow along and contribute [here](https://github.com/supabase/supabase/pull/2281)

**Reply by janaSunrise** [reply]
Hi there! Thank you so much for the Update, Looking forward to it, Really excited! 🙌🏼

**Reply by GoZippy** [reply]
I'm new to this group and just found it - thanks for the update... How can I help with UI?

**Reply by inian** [reply]
You can build upon this PR @GoZippy https://github.com/supabase/supabase/pull/2281
